glslc/src/file_compiler.cc

```diff
@@ -89,28 +125,29 @@ void FileCompiler::AddIncludeDirectory(const std::string& path) {
   include_file_finder_.search_path().push_back(path);
 }
 
-void FileCompiler::SetIndividualCompilationMode() {
+void FileCompiler::SetIndividualCompilationFlags() {
   if (!disassemble_) {
     needs_linking_ = false;
     file_extension_ = ".spv";
   }
 }
 
-void FileCompiler::SetDisassemblyMode() {
-  Compiler::SetDisassemblyMode();
+void FileCompiler::SetDisassemblyFlags() {
+  disassemble_ = true;
```

Comment:
The original FileCompiler inherits from libshaderc_util::compiler, which has an disassemble_ flag, and it is set when libshaderc_util::compiler::SetDisassemblyMode() is called (that is why we can not see any code that sets disassemble_ in the original FileCompiler). But with libshaderc in between, we need to maintain the state flags here, we can not access libshaderc_util::Compiler anymore.